### DISCUSSION TITLE
How to run a single test the correct way?

### DISCUSSION THREAD

**Discussion by velle** [jekyll/jekyll#9549] [Q&A]
What is the right approach for running a single test?

If running all unit tests with `scripts/test` I get this one error every time:

    Failure:
    TestFilters#test_: filters jsonify filter should convert drop to json.  [/home/velle/a/jekyll/test/test_filters.rb:696]
    Minitest::Assertion: --- expected

When I try to run that test alone using `scripts/test test/test_filters.rb` all the tests succeed. 

Am I doing it right?

**Comment by mattr-** [comment]
Yes, you're doing it right. I don't know why that test fails locally and not in CI, but as long as you're not modifying things related to filters, I wouldn't worry too much about it at the moment.
